ckanext/switzerland/tests/test_date_validators.py

```diff
@@ -0,0 +1,81 @@
+# encoding: utf-8
+
+from nose.tools import assert_equals, assert_raises
+from dateutil.parser import parse
+from datetime import datetime
+
+from ckan.plugins.toolkit import get_validator, Invalid
+
+
+class TestOgdchDateStorageValidator(object):
+    def setup(self):
+        self.validator = get_validator('ogdch_date_validator')
+
+    def test_ogdch_date_validator_parseeable_date_to_isodate(self):
+        d = parse('2022-01-02')
+        d_storage = d.isoformat()
+        assert_equals(d_storage, self.validator(d))
+
+    def test_ogdch_date_validator_isodate_to_isodate(self):
+        d = parse('2022-01-02').isoformat()
+        d_storage = d
+        assert_equals(d_storage, self.validator(d))
+
+    def test_ogdch_german_date_to_isodate(self):
+        d = '02.04.2020'
+        d_storage = parse(d, dayfirst=True).isoformat()
+        assert_equals(d_storage, self.validator(d))
+
+    def test_timestamp_to_isodate(self):
+        date_value = parse(str('04.05.2000'), dayfirst=True)
+        epoch = datetime(1970, 1, 1)
+        d = int((date_value - epoch).total_seconds())
+        d_storage = date_value.isoformat()
+        assert_equals(d_storage, self.validator(d))
+
+    def test_ogdch_date_validator_isodate_before_1870_to_isodate(self):
+        d = parse('1655-01-02').isoformat()
+        d_storage = d
+        assert_equals(d_storage, self.validator(d))
+
+    def test_ogdch_date_validator_datetime_to_isodate(self):
+        d = datetime(2022, 1, 2, 0, 0)
+        d_storage = d.isoformat()
+        assert_equals(d_storage, self.validator(d))
+
```

Comment:
Could you please also add test for this kind of case: "2008-09-03T20:56:35.450686Z"?, I am not sure if isoformat parse the tzinfo (a timezone)

Reply:
This test was added. Thanks for this suggestion.